sdks/python/apache_beam/runners/direct/transform_evaluator.py

```diff
@@ -516,7 +524,7 @@ def get_subscription(cls, transform, project, short_topic_name,
 
     sub_client = pubsub.SubscriberClient()
     sub_name = sub_client.subscription_path(
-        project, 'beam_%d_%x' % (int(time.time()), random.randrange(1 << 32)))
+        sub_project, 'beam_%d_%x' % (int(time.time()), random.randrange(1 << 32)))
```

Comment:
We can fall back to topic's project if this fails.

Reply:
Falling back to topic's project may make things less predictable, and failure to create the subscription could be transient and could be because of many reasons. The user should be sure which project it will create the subscription under. My take is that using the topic's project was a mistake from the beginning and it should be fixed. But let me know if you have other thoughts.

Reply:
I _think_ changing that would be a backward incompatible change and might break users. We can keep the existing behavior in place, and add a warning along the lines of this is wrong/deprecated/would change in the future.